### PR TITLE
Desing Patterns by Deyan Sadinov

### PR DESCRIPTION
**Pull request by deyansadinov**
Proxy Pattern - task4

### CODE REVIEW

File: src/task4/proxypattern/proxy/ProxyInteger.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  private RealInteger realInteger;
+
+  public ProxyInteger(int number) {
+    this.number = number;
+  }
+
+  @Override
+  public void printNumber() {
+    if (realInteger == null){
+      realInteger = new RealInteger(number);
```
Comment (mgenov) [inline src/task4/proxypattern/proxy/ProxyInteger.java]: I would like to see this pattern with low coupled dependency.

File: src/task4/proxypattern/proxy/RealInteger.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class RealInteger implements Integer {
+
+  private final int number;
+
+  public RealInteger(int number) {
+    this.number = number;
+    loadFromDisk(number);
```
Comment (mgenov) [inline src/task4/proxypattern/proxy/RealInteger.java]: why you have added this private method ?

Comment (deyansadinov) [inline src/task4/proxypattern/proxy/RealInteger.java]: Yes this method is completely unnecessary. I added a system out which print the loading number

File: src/task4/proxypattern/proxy/ProxyInteger.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package task4.proxypattern.proxy;
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class ProxyInteger implements Integer {
+
+  private final int number;
```
Comment (mgenov) [inline src/task4/proxypattern/proxy/ProxyInteger.java]: This doesn't look like a proxy. Still something is missing here.

Comment (deyansadinov) [inline src/task4/proxypattern/proxy/ProxyInteger.java]: In the ProxyInteger class I make instance of RealInteger

**Pull request closed without merging**